boefjes/plugins/kat_shodan/main.py

```diff
@@ -8,10 +9,15 @@
 
 
 def run(boefje_meta: BoefjeMeta) -> Tuple[BoefjeMeta, Union[bytes, str]]:
-
     api = shodan.Shodan(getenv("SHODAN_API"))
     input_ = boefje_meta.arguments["input"]
     ip = input_["address"]
-    results = api.host(ip)
+    results = {}
+    try:
+        results = api.host(ip)
+    except shodan.APIError as e:
+        if e.args[0] != "No information available for that IP.":
+            raise
+        logging.info(e)
```

Comment:
Thanks for the suggestion! Unfortunately Shodan's `APIError` doesn't provide much detail. But I was wondering, wouldn't it be better to log every error? e.g.:

```suggestion
        logging.error("API error", exc_info=e)
```

Unless you really want to error out specifically on "No information available for that IP.", then your suggestion should work. With only a small modification (`info` to `error`):
```suggestion
        if e.args[0] != "No information available for that IP.":
            raise
        logging.error("API error", exc_info=e)
```

Reply:
Personally, I would say that "No information available for that IP." from Shodan would be something you want to know as informational information, or maybe there are use cases where it would be a warning, but I would not classify it as an error in the application.

Regarding "logging every error", the current behaviour is that any Python error, such as the traceback above, ends up verbatim in the boefjes log. This generic boefjes-behaviour is taken of by the `raise` statement for the case where something did actually go wrong (instead of  "No information available for that IP."). Maybe there are also other API errors next to the "No information available for that IP.", that would also need a different treatment than crashing the whole boefje but I have not encountered those yet :).

Reply:
agreed, this is what I would expect from a Boefje.

Reply:
@zcrt Can you please sign the CLA?